app/main.py

```diff
@@ -1,3 +1,70 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> float:
+        self.km = km
+
+    def __str__(self) -> float:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> float:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: float) -> float:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return NotImplemented
+
+    def __iadd__(self, other: float) -> float:
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (int, float)):
+            self.km += other
+        else:
+            return NotImplemented
+        return self
+
+    def __mul__(self, factor: float) -> float:
```

Comment:
The `__mul__` method should not have a return type annotation of `float`. It should be `Distance` since it returns a new `Distance` object.